cql3/statements/create_aggregate_statement.cc

```diff
@@ -22,7 +22,7 @@ namespace cql3 {
 
 namespace statements {
 
-shared_ptr<db::functions::function> create_aggregate_statement::create(query_processor& qp, db::functions::function* old) const {
+seastar::future<shared_ptr<db::functions::function>> create_aggregate_statement::create(query_processor& qp, db::functions::function* old) const {
```

Comment:
When converting non-future to future there is always a risk of introducing races - now two such statements can execute concurrently.

We can assume Raft, so if Raft already serializes execution, we're fine. I don't have any concrete reasons to think  there's a race here, but I'm not too familiar with the code path, so please double check it.

Reply:
I examined the new futures again, but couldn't find any races that could be caused by them. A potential race that looked like it might appear was between replacing and dropping the UDF, but it looks like we handle it well enough already.